test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/MvcEndpointDataSourceTests.cs

```diff
@@ -152,16 +152,22 @@ private static TheoryData GetSingleActionData(bool isConventionalRouting)
                     {"{CONTROLLER}/{ACTION}/{id?}", new[] { "TestController/TestAction/{id?}" }},
                     {"{controller}/{action=TestAction}", new[] { "TestController", "TestController/TestAction" }},
-                    {"{controller}/{action=TestAction}/{id?}", new[] { "TestController", "TestController/TestAction/{id?}" }},
-                    {"{controller=TestController}/{action=TestAction}/{id?}", new[] { "", "TestController", "TestController/TestAction/{id?}" }},
+                    {"{controller}/{action=TestAction}/{id?}", new[] { "TestController/TestAction/{id}", "TestController", "TestController/TestAction" }},
+                    {"{controller}/{action=TestAction}/{id?}/{more}", new[] { "TestController/TestAction/{id?}/{more}" }},
+                    {"{controller=TestController}/{action=TestAction}/{id?}", new[] { "TestController/TestAction/{id}", "", "TestController", "TestController/TestAction" }},
+                    {"{controller=TestController}/{action=TestAction}/{id?}/{more?}", new[] { "TestController/TestAction/{id}/{more}", "TestController/TestAction/{id}", "", "TestController", "TestController/TestAction" }},
                     {"{controller}/{action}/{*catchAll}", new[] { "TestController/TestAction/{*catchAll}" }},
                     {"{controller}/{action=TestAction}/{*catchAll}", new[] { "TestController", "TestController/TestAction/{*catchAll}" }},
-                    {"{controller}/{action=TestAction}/{id?}/{*catchAll}", new[] { "TestController", "TestController/TestAction/{id?}/{*catchAll}" }},
+                    {"{controller}/{action=TestAction}/{id?}/{*catchAll}", new[] { "TestController/TestAction/{id}/{*catchAll}", "TestController", "TestController/TestAction" }},
+                    {"{controller}/{action=TestAction}/{id?}/{**catchAll}", new[] { "TestController/TestAction/{id}/{**catchAll}", "TestController", "TestController/TestAction" }},
                     {"{controller}/{action}.{ext?}", new[] { "TestController/TestAction.{ext?}" }},
-                    {"{controller}/{action=TestAction}.{ext?}", new[] { "TestController", "TestController/TestAction.{ext?}" }},
-					{"{controller:upper-case}/{action=TestAction}.{ext?}", new[] { "TESTCONTROLLER", "TESTCONTROLLER/TestAction.{ext?}" }},
+                    {"{controller}/{action=TestAction}.{ext?}", new[] { "TestController/TestAction.{ext}", "TestController", "TestController/TestAction" }},
+                    {"{controller}/{action=TestAction}.{ext?}/{more?}", new[] { "TestController/TestAction.{ext}/{more}", "TestController/TestAction.{ext}", "TestController", "TestController/TestAction" }},
+                    {"{controller}/{action=TestAction}.{ext?}/{more}", new[] { "TestController/TestAction.{ext?}/{more}" }},
+                    {"{controller:upper-case}/{action=TestAction}.{ext?}", new[] { "TESTCONTROLLER/TestAction.{ext}", "TESTCONTROLLER", "TESTCONTROLLER/TestAction" }},
```

Comment:
I'd like to also see a case with complex segments with optional parameters. `{x?}-{y?}`

Reply:
https://github.com/aspnet/Mvc/pull/8616#discussion_r226484017